Sources/Core/Shared/GroupOperation.swift

```diff
@@ -225,12 +305,41 @@ public class GroupOperation: Operation, OperationQueueDelegate {
     }
 
     public func operationQueue(queue: OperationQueue, didFinishOperation operation: NSOperation, withErrors errors: [ErrorType]) {
+        guard queue === self.queue else { return }
 
         if operation === finishingOperation {
             finish(fatalErrors)
             queue.suspended = true
         }
     }
+
+    public func operationQueue(queue: OperationQueue, willProduceOperation operation: NSOperation) {
```

Comment:
I really like this addition. We previously add some (poor in hindsight) code in `OperationQueue` to watch out for this. But this is much nicer.